src/components/AnnotationViewer.vue

```diff
@@ -1047,6 +1070,391 @@ export default class AnnotationViewer extends Vue {
     });
   }
 
+  // Helper function to find connected components using Union-Find
+  private findConnectedComponents(
+    connections: IAnnotationConnection[],
+  ): { annotations: Set<string>; connections: IAnnotationConnection[] }[] {
+    // Simple Union-Find implementation
+    const parent = new Map<string, string>();
+
+    // Find with path compression
+    function find(x: string): string {
+      if (!parent.has(x)) {
+        parent.set(x, x);
+      }
+      if (parent.get(x) !== x) {
+        parent.set(x, find(parent.get(x)!));
+      }
+      return parent.get(x)!;
+    }
+
+    // Union operation
+    function union(x: string, y: string): void {
+      parent.set(find(x), find(y));
+    }
+
+    // Process all connections
+    connections.forEach((conn) => {
+      union(conn.parentId, conn.childId);
+    });
+
+    // Group annotations and their connections by root
+    const components = new Map<
+      string,
+      {
+        annotations: Set<string>;
+        connections: IAnnotationConnection[];
+      }
+    >();
+
+    // Initialize components with annotations
+    parent.forEach((_, node) => {
+      const root = find(node);
+      if (!components.has(root)) {
+        components.set(root, {
+          annotations: new Set(),
+          connections: [],
+        });
+      }
+      components.get(root)!.annotations.add(node);
+    });
+
+    // Add connections to their respective components
+    connections.forEach((conn) => {
+      const root = find(conn.parentId); // Could use either parent or child
+      components.get(root)!.connections.push(conn);
+    });
+
+    return Array.from(components.values());
+  }
+
+  drawTimelapseConnectionsAndCentroids() {
+    // Remove all previous tracks and centroids
+    this.timelapseLayer.removeAllAnnotations();
+    this.timelapseTextLayer.features([]);
+
+    if (!this.showTimelapseMode) {
+      // Redraw to ensure we clear the layers. I found that this was necessary for the timelapseTextLayer to clear.
+      this.timelapseLayer.draw();
+      this.timelapseTextLayer.draw();
+      return;
+    }
+
+    // Only render tracks and annotations within this time window
+    const timelapseModeWindow = this.timelapseModeWindow;
+    const currentTime = this.time;
+    const timelapseTags = this.store.timelapseTags;
+
+    // Get connected components to find each individual track
+    const components = this.findConnectedComponents(this.annotationConnections);
```

Comment:
Probably best to filter connections by the current frame location first, then do the call to findConnectedComponents

Reply:
Actually, look around in the code for a filteredAnnotation or displayedAnnotation list somewhere in annotation.ts

Reply:
That was a bit more irritating than anticipated :)